14/umbraco-cms/tutorials/starter-kit/lessons/2-add-a-blog-post-publication-date/part-3.md

```diff
@@ -10,22 +10,38 @@ Finally, in Part Three we shall change the blog listing.
 
     * Select *Default.cshtml*.
 
-2. Scroll down to find the `blogpost-date` element and change it to use a nicely formatted Publication Date, i.e.:
+2. Scroll down to find the `foreach` loop.
+3. Declare a new `publicationDate` variable as the first thing with the loop:
+
+    ```csharp
+    var publicationDate = post.Value<DateTime>("publicationDate");
+    ```
+
+4. Locate the `blogpost-date` element a bit further down and change it to use the new variable:
 
     ```html
-    <small class="blogpost-date">@post.PublicationDate.ToLongDateString()</small>
+    <small class="blogpost-date">@publicationDate.ToLongDateString()</small>
     ```
 
-3. Further up in the view, you also need to redefine the `blogposts` variable:
+    * The `ToLongDateString()` method ensures a clean looking date format.
+
+5. Redefine the `blogposts` variable before the first `div` tag - this will be used for sorting the posts:
 
     ```csharp
-    var blogposts = startNode.Children.OrderByDescending(x => x.Value<DateTime>("PublicationDate")).ToList();
+    @{
+        var blogposts = Model.BlogPosts.OrderByDescending(x => x.Value<DateTime>("publicationDate")).ToList();
+    }
     ```
 
     * Because we are sorting by a custom property we need to use the generic `Value` method.
```

Comment:
```suggestion
    * Because we are sorting by a custom property we need to use the generic `Value` method.
    
{% hint style="info" %}
You can use **Query builder** to construct your queries in a more structured and reusable manner. Use the `UmbracoHelper` or the `IPublishedContentQuery` interface to build queries dynamically. For more information, see the [Querying & Models](https://docs.umbraco.com/umbraco-cms/reference/querying) article.
{% endhint %}
```

Reply:
👏